### DISCUSSION TITLE
Using rectangular patches as grid lines, the rendering will flicker when the camera zooms out

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#7007] [Q&A]
Zoom in on the camera to view the mesh, and the mesh rendering is normal
![CIavQnm4sq](https://github.com/google/filament/assets/132528822/cfa8f595-f930-4015-837f-cd0f9bdb326a)
Camera zooms out to view the grid, rendering will flicker
![h8w2hEhfe5](https://github.com/google/filament/assets/132528822/91105d8f-4e26-4a7c-9c3b-4805fc04dc98)

**Comment by romainguy** [comment]
It's caused by undersanpling. Make sure you provide or compute mip levels for that texture.

**Reply by romainguy** [reply]
And if the grid is made of a mesh there isn't much you can do except enable MSAA and maybe switch to a different LOD at a certain distance.

**Comment by Haidiye00** [comment]
grid is made of a mesh ，every line is made of two triangles

**Reply by romainguy** [reply]
See my other reply. Use MSAA or a texture. A mesh with thin triangles like this won't work well at a distance (unless you swap meshes based on distance). You could also generate the grid procedurally from inside a material and get a perfect result always.

**Comment by Haidiye00** [comment]
thanks
